evergreen/setup_baas_proxy.sh

```diff
@@ -0,0 +1,263 @@
+fi
+
+function check_port_in_use()
+{
+    # Usage: check_port_in_use PORT PORT_NAME
+    port_num="${1}"
+    port_check=$(lsof -P "-i:${port_num}" | grep "LISTEN" || true)
+    if [[ -n "${port_check}" ]]; then
+        echo "Error: ${2} port (${port_num}) is already in use"
+        echo -e "${port_check}"
+        exit 1
+    fi
+}
+
+# Check the baas proxy listen and Toxiproxy config port availability first
+check_port_in_use "${LISTEN_PORT}" "baas proxy"
+check_port_in_use "${CONFIG_PORT}" "Toxiproxy config"
+
+[[ -d "${WORK_PATH}" ]] || mkdir -p "${WORK_PATH}"
+pushd "${WORK_PATH}" > /dev/null
+
+PROXY_CFG_FILE="${WORK_PATH}/baas_proxy.json"
+PROXY_LOG="${WORK_PATH}/baas_proxy.log"
+PROXY_PID_FILE="${WORK_PATH}/baas_proxy.pid"
+PROXY_STOPPED_FILE="${WORK_PATH}/baas_proxy_stopped"
+BAAS_STOPPED_FILE="${BAAS_PATH}/baas_stopped"
+
+# Remove some files from a previous run if they exist
+if [[ -f "${CONFIG_FILE}" ]]; then
+    rm -f "${CONFIG_FILE}"
+fi
+if [[ -f "${PROXY_LOG}" ]]; then
+    rm -f "${PROXY_LOG}"
+fi
+if [[ -f "${PROXY_PID_FILE}" ]]; then
+    rm -f "${PROXY_PID_FILE}"
+fi
+
+if [[ -f "${PROXY_STOPPED}" ]]; then
+    rm -f "${PROXY_STOPPED}"
+fi
+
+# Set up the cleanup function that runs at exit and stops the toxiproxy server
+trap 'on_exit' EXIT
+
+function on_exit()
+{
+    # Usage: on_exit
+    # Toxiproxy is being stopped (or never started), create a 'baas-proxy-stopped' file
+    touch "${PROXY_STOPPED_FILE}" || true
+
+    proxy_pid=
+    if [[ -f "${PROXY_PID_FILE}" ]]; then
+        proxy_pid="$(< "${PROXY_PID_FILE}")"
+    fi
+
+    if [[ -n "${proxy_pid}" ]]; then
+        echo "Stopping baas proxy ${proxy_pid}"
+        kill "${proxy_pid}" || true
+        echo "Waiting for baas proxy to stop"
+        wait
+        rm -f "${PROXY_PID_FILE}" || true
+    fi
+}
+
+case $(uname -s) in
+    Darwin)
+        if [[ "$(uname -m)" == "arm64" ]]; then
+            export GOARCH=arm64
+            GO_URL="https://s3.amazonaws.com/static.realm.io/evergreen-assets/go1.19.3.darwin-arm64.tar.gz"
+            # Go's scheduler is not BIG.little aware, and by default will spawn
+            # threads until they end up getting scheduled on efficiency cores,
+            # which is slower than just not using them. Limiting the threads to
+            # the number of performance cores results in them usually not
+            # running on efficiency cores. Checking the performance core count
+            # wasn't implemented until the first CPU with a performance core
+            # count other than 4 was released, so if it's unavailable it's 4.
+            GOMAXPROCS="$(sysctl -n hw.perflevel0.logicalcpu || echo 4)"
+            export GOMAXPROCS
+        else
+            export GOARCH=amd64
+            GO_URL="https://s3.amazonaws.com/static.realm.io/evergreen-assets/go1.19.1.darwin-amd64.tar.gz"
+        fi
+    ;;
+    Linux)
+        GO_URL="https://s3.amazonaws.com/static.realm.io/evergreen-assets/go1.19.1.linux-amd64.tar.gz"
```

Comment:
are these intentionally using different patch versions than above?

Reply:
I just reused the go versions from the `install_baas.sh` script since we have these binaries readily available. Right now, it is only being run on linux, but it has been tested on both arm and x86_64 macOS as well.